src/foam/dao/jdbc/AbstractJDBCDAO.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.dao.jdbc;
+
+import foam.dao.AbstractDAO;
+
+import foam.core.*;
+
+import javax.sql.DataSource;
+import java.sql.SQLException;
+import java.sql.*;
+
+import java.lang.Exception;
+import java.util.*;
+
+public abstract class AbstractJDBCDAO extends AbstractDAO{
+
+  /** Holds the relevant properties (column names) of the table */
+  protected List<PropertyInfo> properties_;
+
+  protected ThreadLocal<StringBuilder> threadLocalBuilder_ = new ThreadLocal<StringBuilder>(){
+
+    @Override
+    protected StringBuilder initialValue() {
+      return new StringBuilder();
+    }
+
+    @Override
+    public StringBuilder get() {
+      StringBuilder builder = super.get();
+      builder.setLength(0);
+      return builder;
+    }
+
+  };
+
+  protected String tableName_;
+
+  /** Holds a reference to the connection pool ( .getConnection() ) */
+  protected static DataSource dataSource_;
+
+  /**
+   * Create the table in the database and return true if it doesn't already exist otherwise it does nothing and returns false
+   * @param of ClassInfo
+   */
+  public abstract boolean createTable(X x, ClassInfo of);
+
+  public AbstractJDBCDAO(X x, ClassInfo of, String poolName) throws java.sql.SQLException, ClassNotFoundException {
```

Comment:
I don't think the DAO should need to know the poolName. Can this class get access the dataSource from the context and not know what pool it is using?